.github/workflows/build_status.yml

```diff
@@ -7,11 +7,12 @@ on:
 jobs:
   build:
     name: Unit tests
-    runs-on: ubuntu-18.04
     strategy:
       max-parallel: 4
       matrix:
         python-version: [2.7]
+        os: [ubuntu-18.04, macos-latest]
+    runs-on: ${{ matrix.os }}
```

Comment:
As noted in the discussion #9 - the build_status workflow doesn't need to do a full matrix - it just needs a single build as a check that nothing obviously bad has gone wrong in the merge. That's probably an argument for removing the matrix from the build_status workflow config - I'll admit I was lazy and did a copy-paste of the ci.yml config to bootstrap build_status.yml.

Reply:
+1

Personally, I'd prefer the following:

- Use `on: pull_request` to do the full build (the way we're doing it now)
- Cache the results by the Git `tree` (not commit ID, but tree ID, which would mean any `git push` that doesn't change the files will re-use the result already generated for the tree ID)
- Use `on: push` to `master` to do the full build, but since this would be cached by git tree ID, it would be a no-op

I don't have any code handy that would do this, so I'll do it the way you're suggesting :)